stumpui/windows.lisp

```diff
@@ -0,0 +1,215 @@
+  (if (stringp value)
+      (call-next-method (stumpwm:split-string value (string #\Newline))
+                        window)
+      (call-next-method)))
+
+(defmethod initialize-instance :after ((window text-window)
+                                       &key &allow-other-keys)
+  (let* ((screen (window-screen window))
+         (xwin (window-xwin window))
+         (ccontext (stumpwm::make-ccontext
+                    :screen screen
+                    :win xwin
+                    :gc (xlib:create-gcontext :drawable xwin))))
+    (setf (%text-window-ccontext window)
+          ccontext
+          (text-window-foreground-color window)
+          (stumpwm::screen-fg-color screen)
+          (text-window-background-color window)
+          (stumpwm::screen-bg-color screen)
+          (text-window-border-color window)
+          (stumpwm::screen-border-color screen))
+    (stumpwm::reset-color-context ccontext)))
+
+;;; Style
+
+(defmethod (setf text-window-foreground-color) (color (window text-window))
+  (let* ((screen (window-screen window))
+         (ccontext (text-window-ccontext window))
+         (gcontext (stumpwm::ccontext-gc ccontext)))
+    (setf (xlib:gcontext-foreground gcontext) color
+          (stumpwm::ccontext-default-fg ccontext) color
+          ;; Default the bright foreground color to the normal foreground in
+          ;; case we fail to get the proper bright one.
+          (stumpwm::ccontext-default-bright ccontext) color)
+    ;; KLUDGE: We can calculate the default bright color from the foreground
+    ;; color, but that's a PIXEL, not a COLOR. To get the COLOR, we need to
+    ;; look it up.
+    (let* ((allocated-colors (xlib:query-colors
+                              (xlib:screen-default-colormap
+                               (stumpwm:screen-number screen))
+                              (list color)))
+           (bright (and (plusp (length allocated-colors))
+                        (elt allocated-colors 0))))
+      (when bright
+        (stumpwm::adjust-color bright 0.25)
+        (setf (stumpwm::ccontext-default-bright ccontext)
+              (stumpwm::alloc-color screen bright))))))
+
+(defmethod (setf text-window-background-color) (color (window text-window))
+  (let* ((ccontext (text-window-ccontext window))
+         (gcontext (stumpwm::ccontext-gc ccontext))
+         (xwin (window-xwin window)))
+    (setf (xlib:gcontext-background gcontext) color
+          (xlib:window-background xwin) color
+          (stumpwm::ccontext-default-bg ccontext) color)))
+
+(defmethod (setf text-window-border-color) (color (window text-window))
+  (setf (xlib:window-border (window-xwin window))
+        color))
+
+(defmethod (setf text-window-border-width) (width (window text-window))
+  (setf (xlib:drawable-border-width (window-xwin window))
+        width))
+
+;;; Timed window --------------------------------------------------------------
+;;; The TIMED-WINDOW class implements the Timed window protocol.
```

Comment:
If these are pop-ups lets call them that: 
https://tronche.com/gui/x/icccm/sec-4.html#s-4.1.10